File: ProgrammingChallenge/app/src/main/java/com/gabrielbmoro/programmingchallenge/domain/usecase/PopularMoviesUseCase.kt
package com.gabrielbmoro.programmingchallenge.domain.usecase

import com.gabrielbmoro.programmingchallenge.domain.model.Page
import com.gabrielbmoro.programmingchallenge.domain.usecase.mapper.MoviesMapper
import com.gabrielbmoro.programmingchallenge.repository.MoviesRepository

class PopularMoviesUseCase(val repository: MoviesRepository) {

    suspend fun execute(pageNumber: Int): Page? {
        return MoviesMapper.mapToPage(repository.getPopularMovies(pageNumber))
    }
}

File: flutter-version/ProgrammingChallenge/movie_db_app/lib/domain/usecase/popular_movies_use_case.dart
import 'package:movie_db_app/domain/model/page.dart';
import 'package:movie_db_app/repository/movie_db_api_repository.dart';

class GetPopularMoviesUseCase {
  MovieDBRepository _repository;

  GetPopularMoviesUseCase(MovieDBRepository repository) {
    _repository = repository;
  }

  Future<Page> execute(int pageNumber) {
    return _repository.getPopularMovies(
      pageNumber: pageNumber,
    );
  }
}
